test-cypress/integration/setup/user-and-access.spec.js

```diff
@@ -59,4 +47,241 @@ describe('User and Access', () => {
         // Date created should be visible
         cy.get('@user').find('.date-created').should('be.visible');
     });
+
+    it('Create user - read-only and verify that security works', () => {
+        cy.wait(500); //sometimes repositories list cannot be fetched.
+        getCreateNewUserButton().click();
+        getUsernameField().type(user);
+        getPasswordField().type(password);
+        getConfirmPasswordField().type(password);
+        getUserRoleRadioButton("roleUser").click();
+        getRepositoryRights(repositoryId, "read").click();
+        getCreateButton()
+            .should('be.visible')
+            .click();
+        findUserInTable(user);
+        getSecuritySwitch().click({force:true});
+        performLogin(user, password);
+        //when logging in with a non admin user you will be redirected to the last page you logged out from and in this case the user does not have rights to view it.
+        cy.url()
+            .should('eq', Cypress.config('baseUrl') + '/users');
+        cy.get('.container-fluid')
+            .should('contain', 'You have no permission to access this functionality with your current credentials');
+        //verify user rights
+        getMenuItem("Repositories")
+            .should('not.be.visible');
+        getMenuItem("Users and Access")
+            .should('not.be.visible');
+        //verify no access message is displayed
+        getNoPermissionMessage("import")
+            .should('be.visible')
+            .and('contain', 'Some functionality is not available because you have no write permission to repository ' + repositoryId);
+        getNoPermissionMessage("connectors")
+            .should('be.visible')
+            .and('contain', 'Some functionality is not available because you have no write permission to repository ' + repositoryId);
+        getNoPermissionMessage("autocomplete")
+            .should('be.visible')
+            .and('contain', 'Some functionality is not available because you have no write permission to repository ' + repositoryId);
+        getNoPermissionMessage("rdfrank")
+            .should('be.visible')
+            .and('contain', 'Some functionality is not available because you have no write permission to repository ' + repositoryId);
+        cy.visit('/users');
+        getUserLogoutButton().click();
+        performLogin("admin", adminPassword);
+        getDeleteUserButton(user).click();
+        getDeleteUserConfirmButton().click();
+        getSecuritySwitch().click({force:true});
```

Comment:
This could be extracted in a function for deleting user and reused as well.

Reply:
Fixed with commit d5a40ea358aa33047d813e1f1b223285da550018